lib/const.rb

```diff
@@ -87,74 +87,59 @@
   CREW_KERNEL_VERSION = ENV.fetch('CREW_KERNEL_VERSION', `uname -r`.rpartition('.')[0])
 end
 
-CREW_LIB_PREFIX = "#{CREW_PREFIX}/#{ARCH_LIB}"
-CREW_MAN_PREFIX = "#{CREW_PREFIX}/share/man"
-CREW_LIB_PATH = "#{CREW_PREFIX}/lib/crew/"
-CREW_PACKAGES_PATH = "#{CREW_LIB_PATH}packages/"
-CREW_CONFIG_PATH = "#{CREW_PREFIX}/etc/crew/"
-CREW_META_PATH = "#{CREW_CONFIG_PATH}meta/"
-CREW_BREW_DIR = "#{CREW_PREFIX}/tmp/crew/"
-CREW_DEST_DIR = "#{CREW_BREW_DIR}dest"
-CREW_DLL_PREFIX = "#{CREW_LIB_PREFIX}/wine"
-CREW_DEST_PREFIX = CREW_DEST_DIR + CREW_PREFIX
-CREW_DEST_LIB_PREFIX = CREW_DEST_DIR + CREW_LIB_PREFIX
-CREW_DEST_DLL_PREFIX = CREW_DEST_PREFIX + CREW_DLL_PREFIX
-CREW_DEST_MAN_PREFIX = CREW_DEST_DIR + CREW_MAN_PREFIX
+CREW_LIB_PREFIX      = File.join(CREW_PREFIX, ARCH_LIB)
+CREW_MAN_PREFIX      = File.join(CREW_PREFIX, 'share/man/')
+CREW_LIB_PATH        = File.join(CREW_PREFIX, 'lib/crew/')
+CREW_PACKAGES_PATH   = File.join(CREW_LIB_PATH, 'packages/')
+CREW_CONFIG_PATH     = File.join(CREW_PREFIX, 'etc/crew/')
+CREW_META_PATH       = File.join(CREW_CONFIG_PATH, 'meta/')
+CREW_BREW_DIR        = File.join(CREW_PREFIX, 'tmp/crew/')
+CREW_DEST_DIR        = File.join(CREW_BREW_DIR, 'dest/')
+CREW_DLL_PREFIX      = File.join(CREW_LIB_PREFIX, 'wine/')
+CREW_DEST_PREFIX     = File.join(CREW_DEST_DIR, CREW_PREFIX)
+CREW_DEST_LIB_PREFIX = File.join(CREW_DEST_DIR, CREW_LIB_PREFIX)
+CREW_DEST_DLL_PREFIX = File.join(CREW_DEST_PREFIX, CREW_DLL_PREFIX)
+CREW_DEST_MAN_PREFIX = File.join(CREW_DEST_DIR, CREW_MAN_PREFIX)
 
 # Local constants for contributors.
-repo_root = `git rev-parse --show-toplevel 2> /dev/null`.chomp.to_s
-if repo_root.empty?
-  Dir.chdir '../..' do
-    repo_root = `git rev-parse --show-toplevel 2> /dev/null`.chomp.to_s
-  end
-end
+repo_root = `git rev-parse --show-toplevel 2> /dev/null`.chomp
+repo_root = `cd ../..; git rev-parse --show-toplevel 2> /dev/null`.chomp if repo_root.empty?
```

Comment:
`cd ../..` is a bad assumption.  What if you are in `/` or `/usr` to start?  A better approach would be to `cd #{CREW_LIB_PATH} && ...`.  This should always provide `repo_root` unless someone deliberately deletes the `.git` directory.

Reply:
Also `&&` is better than `;` because it will abort if the directory doesn't exist.

Reply:
This part is modified from the current code (I am not very familiar with the usage of this variable actually😅)

Reply:
It will run inside `CREW_LIB_PATH` now